src/components/SurgeAssetData.js

```diff
@@ -0,0 +1,91 @@
+import SUSDJSON from './json/token_abis/surge_usd_abi.json';
+import SETHJSON from './json/token_abis/surge_eth_abi.json';
+import SBTCJSON from './json/token_abis/surge_btc_abi.json';
+import SADAJSON from './json/token_abis/surge_ada_abi.json';
+import SUSLSJSON from './json/token_abis/surge_useless_abi.json';
+import XUSDJSON from './json/token_abis/surge_xusd_abi.json';
+import BNBxUSDFARM from './json/farm_abis/bnb_xusd_farm_abi.json';
+import SurgeTokensData from './json/surge_tokens.json';
+import SurgeFarmsData from './json/surge_farms.json';
+
+const SurgeTokens = SurgeTokensData;
+const SurgeFarms = SurgeFarmsData;
+const SurgeTokensCache = {};
+const SurgeFarmsCache = {};
+
+export function getSurgeTokensData() {
+	for (const token in SurgeTokens) {
+		let abi = '';
+		switch (SurgeTokens[token]['abi_name']) {
+			case 'SUSDJSON':
+				abi = SUSDJSON;
+				break;
+			case 'SETHJSON':
+				abi = SETHJSON;
+				break;
+			case 'SBTCJSON':
+				abi = SBTCJSON;
+				break;
+			case 'SADAJSON':
+				abi = SADAJSON;
+				break;
+			case 'SUSLSJSON':
+				abi = SUSLSJSON;
+				break;
+			case 'XUSDJSON':
+				abi = XUSDJSON;
+				break;
+			default:
+				abi = '';
+				break;
+		}
```

Comment:
I think this part might me more efficient written as an object like
```javascript
const abiObj = {
    SETHJSON: SETHJSON,
    SBTCJSON: SBTCJSON,
    // etc
}
// then matching like
SurgeTokens[token]['abi'] = abiObj[SurgeTokens[token]['abi_name']]
```
Minor optimization tweak, but will probably be good for future cases (especially when surgeFactory comes around)

Reply:
done